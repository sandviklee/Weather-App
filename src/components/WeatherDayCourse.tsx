import style from "../style/WeatherDayCourse.module.css";

interface Info {
    extremalTemperatures: string;
    weatherCourse: Array<string>;
}

const WeatherDayCourse = () => {
    return (
        <main className={style.main}>
            <div className={style.container}>
                <p>Hei!</p>
            </div>
            <div className={style.container}>
                <p>Hei!</p>
            </div>
            <div className={style.container}>
                <p>Hei!</p>
            </div>
            <div className={style.container}>
                <p>Hei!</p>
            </div>
        </main>
    );
};

export default WeatherDayCourse;
