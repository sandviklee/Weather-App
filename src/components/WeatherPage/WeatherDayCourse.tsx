import style from "./WeatherDayCourse.module.css";

interface DayCourseProps {
    extremalTemperatures: Array<string>;
    weatherCourse: Array<string>;
}

/**
 * This is the Course of day Weather Information component.
 * @summary Renders a component with information about the course of day
 * @param props (extrmalTemperatures, weatherCourse)
 * @returns Weather Course of Day JSX Element
 */
const WeatherDayCourse = (props: DayCourseProps): JSX.Element => {
    return (
        <main className={style.main}>
            <p className={style.extremalTemp}>Maks/Min Temperatur</p>
            <p className={style.temperatures}>
                {props.extremalTemperatures[0]}C° /{" "}
                {props.extremalTemperatures[1]}C°
            </p>

            <div className={style.weathers}>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>NATT</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[0]}.svg`}
                            alt="Morning Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>MORGEN</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[1]}.svg`}
                            alt="Noon Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>AFTEN</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[2]}.svg`}
                            alt="Afternoon Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>KVELD</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[3]}.svg`}
                            alt="Night Weather"
                        />
                    </div>
                </div>
            </div>
        </main>
    );
};

export default WeatherDayCourse;
