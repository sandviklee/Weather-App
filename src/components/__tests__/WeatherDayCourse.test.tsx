import { render, screen } from "@testing-library/react";
import WeatherDayCourse from "../WeatherPage/WeatherDayCourse";

const mockData = {
    extremalTemp: ["10", "2"],
    weatherCourse: ["no_weather", "no_weather", "no_weather", "no_weather"],
};

describe("WeatherDayCourse component", () => {
    test("render correctly", () => {
        render(
            <WeatherDayCourse
                extremalTemperatures={mockData.extremalTemp}
                weatherCourse={mockData.weatherCourse}
            />
        );

        const extremalTempTitle = screen.getByText("Maks/Min Temperatur");
        expect(extremalTempTitle).toBeInTheDocument();
        const extremalTemp = screen.getByText("10C° / 2C°");
        expect(extremalTemp).toBeInTheDocument();
        const imageMorning = screen.getByAltText("Morning Weather");
        expect(imageMorning).toBeInTheDocument();
        const imageNoon = screen.getByAltText("Noon Weather");
        expect(imageNoon).toBeInTheDocument();
        const imageAfternoon = screen.getByAltText("Afternoon Weather");
        expect(imageAfternoon).toBeInTheDocument();
        const imageNight = screen.getByAltText("Night Weather");
        expect(imageNight).toBeInTheDocument();
    });

    test("matches snapshot", () => {
        const { asFragment } = render(
            <WeatherDayCourse
                extremalTemperatures={mockData.extremalTemp}
                weatherCourse={mockData.weatherCourse}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});