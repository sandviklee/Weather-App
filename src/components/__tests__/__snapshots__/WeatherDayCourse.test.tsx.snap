// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherDayCourse component > matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="_main_26373e"
  >
    <p
      class="_extremalTemp_26373e"
    >
      Maks/Min Temperatur
    </p>
    <p
      class="_temperatures_26373e"
    >
      10C° / 2C°
    </p>
    <div
      class="_weathers_26373e"
    >
      <div
        class="_container_26373e"
      >
        <div
          class="_weatherContainer_26373e"
        >
          <p>
            NATT
          </p>
          <img
            alt="Morning Weather"
            class="_weatherImage_26373e"
            src="/project1/yr-icons/no_weather.svg"
          />
        </div>
      </div>
      <div
        class="_container_26373e"
      >
        <div
          class="_weatherContainer_26373e"
        >
          <p>
            MORGEN
          </p>
          <img
            alt="Noon Weather"
            class="_weatherImage_26373e"
            src="/project1/yr-icons/no_weather.svg"
          />
        </div>
      </div>
      <div
        class="_container_26373e"
      >
        <div
          class="_weatherContainer_26373e"
        >
          <p>
            AFTEN
          </p>
          <img
            alt="Afternoon Weather"
            class="_weatherImage_26373e"
            src="/project1/yr-icons/no_weather.svg"
          />
        </div>
      </div>
      <div
        class="_container_26373e"
      >
        <div
          class="_weatherContainer_26373e"
        >
          <p>
            KVELD
          </p>
          <img
            alt="Night Weather"
            class="_weatherImage_26373e"
            src="/project1/yr-icons/no_weather.svg"
          />
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
